### ISSUE TITLE
checkpoints/new_tunedvae

### ISSUE THREAD

**Issue by jaekyeongg** [UCSB-NLP-Chang/DiffSTE#7]
Thank you for sharing nice code and research!!! 
I just would like to know more detail.

When I checked diffste.ckpt, It seem to use tunned vae 'vae': {'normalizer': 0.21966682713, 'pretrained_model_path': 'checkpoints/new_tunedvae'} not original vae from stable diffusion. 
Could you share tunned vae??
Actually When I trained it from scratch with training data, Accuacy was much worse than diffste.ckpt.
Could you share some hint??I should use tunned vae???

Config file in diffste.ckpt :
{'source': 'raw', 'target': 'trainers.CharInpaintModelWrapper', 'pretrained_model_path': 'runwayml/stable-diffusion-inpainting', 'loss_type': 'MaskMSELoss', 'loss_alpha': 5, 'base_learning_rate': 5e-05, 'precision': 16, 'weight_decay': 0.0, 'adam_epsilon': 1e-08, 'freeze_char_embedder': False, 'optimize_vae': False, 'vae': {'normalizer': 0.21966682713, 'pretrained_model_path': 'checkpoints/new_tunedvae'}, 'tokenizer': {'model_max_length': 20}, 'char_tokenizer': {'pretrained_path': 'checkpoints/chartokenizer', 'pad_token': ' ', 'unk_token': ' ', 'model_max_length': 20}, 'char_embedder': {'vocab_size': 95, 'embedding_dim': 32, 'max_length': 20, 'padding_idx': 0, 'attention_head_dim': 2}, 'unet': {'attention_head_dim': {'text': 8, 'char': 2}, 'cross_attention_dim': {'text': 768, 'char': 32}}, 'noise_scheduler': 'diffusers.DDIMScheduler', 'ckpt_dir': 'logs/chardiffusion/2023-02-17T21-17-30_alldata/checkpoints'}

**Comment by Question406** [comment]
Thank you for your interest in our work. 

The tuned VAE is actually included in the released checkpoint. You can access it using model.vae and save its weights using model.vae.save_pretrained("save_path"). 

If training from the original VAE doesn't yield satisfactory results, I think it's definitely worthwhile to fine-tune the VAE on your data.

**Comment by jaekyeongg** [comment]
Thank you for kindly explanation.
But I should load diffste ckpt model first with CharInpaintTrainer Class to model.vae.save_pretrained("save_path").
=> model = CharInpaintTrainer(config=config)
When I load CharInpaintTrainer Class, vae is loaded from runwayml/stable-diffusion-inpainting because there is no fintunned vae like below code. 
print("There is no vae from finetunned.  So loaded original vae from runwayml/stable-diffusion-inpainting")
So original vae was saved with model.vae.save_pretrained("save_path").


class CharInpaintTrainer(pl.LightningModule):
    def __init__(self, config):
        super().__init__()
        self.save_hyperparameters()
        self.config = config

        FIXED_COMMIT_ID = FIXED_COMMIT_IDS.get(
            config.pretrained_model_path, None)
        pretrained_model_path = config.pretrained_model_path
        if self.config.vae is not None and self.config.vae.get(
            "pretrained_model_path", False
        ):
            print("Initialize vae from finetuned...")
            if os.path.exists(self.config.vae.get("pretrained_model_path")):
                self.vae = AutoencoderKL.from_pretrained(
                    self.config.vae.get("pretrained_model_path"),
                )
            else:
                print("There is no vae from finetunned. So loaded original vae from runwayml/stable-diffusion-inpainting")
                self.vae = AutoencoderKL.from_pretrained(
                    pretrained_model_path,
                    subfolder="vae",
                    revision=FIXED_COMMIT_ID,
                )
            self.NORMALIZER = self.config.vae.get("normalizer", 0.18215)
        else:
            print("Initialize vae from pretrained...")
            self.vae = AutoencoderKL.from_pretrained(
                pretrained_model_path,
                subfolder="vae",
                revision=FIXED_COMMIT_ID,
            )
            self.NORMALIZER = 0.18215  # original normalizer

**Comment by Question406** [comment]
This is correct. When you run `CharInpaintTrainer.load_from_checkpoint(ckpt_path)`, a new `CharInpaintTrainer` object is created with randomly initialized weights, then the real model weights from `ckpt_path` will be replaced into the new model. I hope it helps.
